controllers/azuremanagedcontrolplane_controller.go

```diff
@@ -113,6 +113,8 @@ func (amcpr *AzureManagedControlPlaneReconciler) SetupWithManager(ctx context.Co
 // +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status,verbs=get;list;watch
 // +kubebuilder:rbac:groups=resources.azure.com,resources=resourcegroups,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=resources.azure.com,resources=resourcegroups/status,verbs=get;list;watch
+// +kubebuilder:rbac:groups=network.azure.com,resources=privateendpoints,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=network.azure.com,resources=privateendpoints/status,verbs=get;list;watch
```

Comment:
I can remove these annotations from here since adding these is duplicating it in `role.yaml`.
But keeping it here for discussion.

Reply:
I'm overall +1 to keeping this here since it serves as decent documentation even if it makes no tangible difference to the generated RBAC when it's already defined elsewhere.